src/test/java/io/reactivex/rxjava3/internal/operators/flowable/FlowablePublishFunctionTest.java

```diff
@@ -538,4 +539,68 @@ public boolean test(Integer w) throws Exception {
         .test()
         .assertResult(1000000);
     }
+
+    @Test
+    public void noUpstreamCancelOnCasualChainClose() {
+        AtomicBoolean parentUpstreamCancelled = new AtomicBoolean(false);
+        Flowable.range(1, 10)
+                .doOnCancel(new Action() {
```

Comment:
You can use lambdas in tests now.

Reply:
Cool!
Should I change it?

Reply:
Yes.

Reply:
If that's fine, I've refactored whole file to lambdas.